CRM/MembershipExtras/Hook/Post/EntityFinancialTrxn.php

```diff
@@ -41,13 +41,39 @@ public function updatePaymentPlanStatus() {
 
     $newStatus = $this->generatePaymentPlanNewStatus();
     if ($newStatus !== NULL) {
+      if ($newStatus === 'Completed') {
+        $subscriptionLines = $this->getSubscriptionLines();
+        foreach($subscriptionLines as $line) {
+          if (!empty($line['start_date']) && empty($line['end_date'])) {
+            civicrm_api3('ContributionRecurLineItem', 'create', [
+              'id' => $line['id'],
+              'end_date' => $this->recurContribution['end_date'],
+            ]);
+          }
+        }
+      }
+
       civicrm_api3('ContributionRecur', 'create', [
         'id' => $this->recurContribution['id'],
         'contribution_status_id' => $newStatus,
       ]);
     }
   }
 
+  /**
+   * Returns LineItems associated to a recurring contribution
+   * 
+   * @return array
+   */
+  private function getSubscriptionLines() {
+    $lines = civicrm_api3('ContributionRecurLineItem', 'get', [
+      'sequential' => 1,
+      'contribution' => $this->recurContribution['id'],
+    ]);
+
+    return $lines;
```

Comment:
have you tested this code ? because it would not work like this, civicrm API return the API rows under the 'values' array key. so you need to adjust this method as : 

```
   $subscriptionLinesResponse= civicrm_api3('ContributionRecurLineItem', 'get', [
      'sequential' => 1,
      'contribution' => $this->recurContribution['id'],
      'options' => ['limit' => 0],
    ]);

    $subscriptionLines = [];
    if (!empty($subscriptionLinesResponse['values'])){ 
      $subscriptionLines = $subscriptionLinesResponse['values'];
    }

    return  $subscriptionLines;
```